provider/lib.go

```diff
@@ -17,3 +21,62 @@ func FilterFilePattern(regex string, filepath string) (bool, error) {
 	return filebool.Match([]byte(filepath)), nil
 
 }
+
+func FindFilesMatchingPattern(root, pattern string) ([]string, error) {
+	var regex *regexp.Regexp
+	// if the regex doesn't compile, we'll default to using filepath.Match on the pattern directly
+	regex, _ = regexp.Compile(pattern)
+	matches := []string{}
+	err := filepath.WalkDir(root, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		var matched bool
+		if regex != nil {
+			matched = regex.MatchString(d.Name())
+		} else {
+			// TODO(fabianvf): is a fileglob style pattern sufficient or do we need regexes?
+			matched, err = filepath.Match(pattern, d.Name())
+			if err != nil {
+				return err
+			}
+		}
+		if matched {
+			matches = append(matches, path)
+		}
+		return nil
+	})
+	return matches, err
+}
+
+func Getfiles(configLocation string, filepaths []string) ([]string, error) {
+	var xmlFiles []string
+
+	if len(filepaths) == 1 {
```

Comment:
I am having a tough time seeing why a single filepath would be different here.

Reply:
what do you mean different?

Reply:
I forgot about the edge case that we talked about the other day, where when there is a single file path, it could be that it is the rendered output from the template, and it could be the case that it is a space-separated list.

I think that we can resolve this one, my bad folks